### ISSUE TITLE
As I reported in issue # 20, I am creating a new issue. Compilation error in kernel Nethunter.

### ISSUE THREAD

**Issue by pe3p** [kimocoder/realtek_rtwifi#21]
I used this kernel source:
git clone https://github.com/Re4son/android_kernel_lge_hammerhead -b nethunter-7.1
This toolchain:
git clone https://android.googlesource.com/platform/prebuilts/gcc/linux-x86/arm/arm-eabi-4.8
I maked next:
export PATH=~/arm-eabi-4.8/bin:$PATH
export ARCH=arm
export CROSS_COMPILE=arm-eabi-
cd ~/android_kernel_lge_hammerhead
make defconfig nethunter_defconfig
make menuconfig //The necessary drivers were displayed in the menuconfig and I selected them
make -j$(nproc)
And at this point the compilation was interrupted: (everything else is without errors)
`CC      drivers/net/wireless/bcmdhd/linux_osl.o
  CC      drivers/net/wireless/realtek/rtl818x/rtl8187/dev.o
  CC      drivers/thermal/msm_thermal.o
  CC      drivers/net/wireless/bcmdhd/bcmsdh_sdmmc.o
  CC      drivers/net/wireless/ath/ath9k/ani.o
  CC      drivers/thermal/msm_thermal-dev.o
  CC      drivers/net/wireless/bcmdhd/dhd_custom_gpio.o
  CC      drivers/net/wireless/ath/ath9k/mac.o
  CC      drivers/net/wireless/realtek/rtl818x/rtl8187/rtl8225.o
  CC      drivers/thermal/msm8974-tsens.o
  CC      drivers/net/wireless/bcmdhd/sbutils.o
  CC      drivers/net/wireless/ath/ath9k/ar9002_mac.o
  CC      drivers/thermal/qpnp-temp-alarm.o
  CC      drivers/net/wireless/bcmdhd/wldev_common.o
  CC      drivers/net/wireless/realtek/rtl818x/rtl8187/leds.o
  CC      drivers/thermal/qpnp-adc-tm.o
  CC      drivers/net/wireless/realtek/rtl818x/rtl8187/rfkill.o
  CC      drivers/net/wireless/bcmdhd/wl_android.o
  CC      drivers/net/wireless/ath/ath9k/ar9003_mac.o
  LD      drivers/net/wireless/realtek/rtl818x/rtl8187/rtl8187.o
  LD      drivers/net/wireless/realtek/rtl818x/rtl8187/built-in.o
  LD      drivers/net/wireless/realtek/rtl818x/built-in.o
  CC      drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.o
  LD      drivers/thermal/built-in.o
  CC      drivers/tty/tty_io.o
drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c: In function 'rtl8xxxu_gen1_config_channel':
drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c:1153:20: error: 'struct ieee80211_conf' has no member named 'chandef'
  channel = hw->conf.chandef.chan->hw_value;
                    ^
drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c:1155:18: error: 'struct ieee80211_conf' has no member named 'chandef'
  switch (hw->conf.chandef.width) {
                  ^
drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c:1156:7: error: 'NL80211_CHAN_WIDTH_20_NOHT' undeclared (first use in this function)
  case NL80211_CHAN_WIDTH_20_NOHT:
       ^
drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c:1156:7: note: each undeclared identifier is reported only once for each function it appears in
drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c:1159:7: error: 'NL80211_CHAN_WIDTH_20' undeclared (first use in this function)
  case NL80211_CHAN_WIDTH_20:
       ^
drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c:1175:7: error: 'NL80211_CHAN_WIDTH_40' undeclared (first use in this function)
  case NL80211_CHAN_WIDTH_40:
       ^
drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c:1176:15: error: 'struct ieee80211_conf' has no member named 'chandef'
   if (hw->conf.chandef.center_freq1 >
               ^
drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c:1177:15: error: 'struct ieee80211_conf' has no member named 'chandef'
       hw->conf.chandef.chan->center_freq) {
               ^
drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c:1257:15: error: 'struct ieee80211_conf' has no member named 'chandef'
   if (hw->conf.chandef.width == NL80211_CHAN_WIDTH_40)
               ^
drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c: In function 'rtl8xxxu_gen2_config_channel':
drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c:1277:20: error: 'struct ieee80211_conf' has no member named 'chandef'
  channel = hw->conf.chandef.chan->hw_value;
                    ^
drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c:1282:18: error: 'struct ieee80211_conf' has no member named 'chandef'
  switch (hw->conf.chandef.width) {
                  ^
drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c:1283:7: error: 'NL80211_CHAN_WIDTH_20_NOHT' undeclared (first use in this function)
  case NL80211_CHAN_WIDTH_20_NOHT:
       ^
drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c:1286:7: error: 'NL80211_CHAN_WIDTH_20' undeclared (first use in this function)
  case NL80211_CHAN_WIDTH_20:
       ^
drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c:1303:7: error: 'NL80211_CHAN_WIDTH_40' undeclared (first use in this function)
  case NL80211_CHAN_WIDTH_40:
       ^
drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c:1306:15: error: 'struct ieee80211_conf' has no member named 'chandef'
   if (hw->conf.chandef.center_freq1 >
               ^
drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c:1307:15: error: 'struct ieee80211_conf' has no member named 'chandef'
       hw->conf.chandef.chan->center_freq) {
               ^
drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c:1349:7: error: 'NL80211_CHAN_WIDTH_80' undeclared (first use in this function)
  case NL80211_CHAN_WIDTH_80:
       ^
drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c:1380:18: error: 'struct ieee80211_conf' has no member named 'chandef'
   switch(hw->conf.chandef.width) {
                  ^
drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c: In function 'rtl8xxxu_bss_info_changed':
drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c:4557:11: error: 'struct ieee80211_sta' has no member named 'vht_cap'
    if (sta->vht_cap.vht_supported)
           ^
drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c: In function 'rtl8xxxu_alloc_tx_urb':
drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c:4705:2: error: implicit declaration of function 'list_first_entry_or_null' [-Werror=implicit-function-declaration]
  tx_urb = list_first_entry_or_null(&priv->tx_urb_free_list,
  ^
drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c:4706:8: error: expected expression before 'struct'
        struct rtl8xxxu_tx_urb, list);
        ^
drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c: In function 'rtl8xxxu_fill_txdesc_v1':
drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c:4835:2: error: implicit declaration of function 'IEEE80211_SEQ_TO_SN' [-Werror=implicit-function-declaration]
  seq_number = IEEE80211_SEQ_TO_SN(le16_to_cpu(hdr->seq_ctrl));
  ^
drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c: At top level:
drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c:5030:11: warning: 'struct ieee80211_tx_control' declared inside parameter list [enabled by default]
error, forbidden warning: rtl8xxxu_core.c:5030
make[5]: *** [scripts/Makefile.build:307: drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.o] Error 1
make[4]: *** [scripts/Makefile.build:443: drivers/net/wireless/realtek/rtl8xxxu] Error 2
make[3]: *** [scripts/Makefile.build:443: drivers/net/wireless/realtek] Error 2
make[3]: *** Waiting for unfinished jobs....
  CC      drivers/tty/n_tty.o
  CC      drivers/net/wireless/ath/ath9k/ar9003_eeprom.o
  CC      drivers/net/wireless/bcmdhd/wl_cfg80211.o
  CC      drivers/tty/tty_ioctl.o
  CC      drivers/net/wireless/ath/ath9k/ar9003_paprd.o
  CC      drivers/tty/tty_ldisc.o
  CC      drivers/tty/tty_buffer.o
  CC      drivers/tty/tty_port.o
  CC      drivers/net/wireless/ath/ath9k/btcoex.o
  CC      drivers/tty/tty_mutex.o
  CC      drivers/tty/pty.o
  CC      drivers/tty/tty_audit.o
  CC      drivers/net/wireless/ath/ath9k/ar9003_mci.o
  CC      drivers/tty/sysrq.o
  LD      drivers/tty/ipwireless/built-in.o
  CC      drivers/tty/serial/serial_core.o
  CC      drivers/tty/vt/vt_ioctl.o
  LD      drivers/net/wireless/ath/ath9k/ath9k_hw.o
  LD      drivers/net/wireless/ath/ath9k/ath9k_common.o
  LD      drivers/net/wireless/ath/ath9k/ath9k_htc.o
  LD      drivers/net/wireless/ath/ath9k/built-in.o
  CC      drivers/net/wireless/ath/carl9170/main.o
  CC      drivers/tty/serial/msm_serial_hs.o
  CC      drivers/tty/vt/vc_screen.o
  CC      drivers/net/wireless/ath/carl9170/usb.o
  CC      drivers/tty/vt/selection.o
  CC      drivers/tty/vt/keyboard.o
  CC      drivers/net/wireless/ath/carl9170/cmd.o
  CC      drivers/tty/serial/msm_serial_hs_lite.o
  CC      drivers/net/wireless/ath/carl9170/mac.o
  CC      drivers/net/wireless/bcmdhd/wl_cfgp2p.o
  CC      drivers/tty/vt/consolemap.o
  CC      drivers/net/wireless/ath/carl9170/phy.o
  LD      drivers/tty/serial/built-in.o
  CC      drivers/net/wireless/ath/carl9170/led.o
  CC      drivers/net/wireless/ath/carl9170/fw.o
  CONMK   drivers/tty/vt/consolemap_deftbl.c
  CC      drivers/tty/vt/vt.o
  CC      drivers/net/wireless/ath/carl9170/tx.o
  CC      drivers/net/wireless/ath/carl9170/rx.o
  CC      drivers/net/wireless/bcmdhd/wl_linux_mon.o
  CC      drivers/net/wireless/bcmdhd/dhd_cfg80211.o
  LD      drivers/net/wireless/ath/carl9170/carl9170.o
  CC      drivers/net/wireless/bcmdhd/wl_cfgvendor.o
  LD      drivers/net/wireless/ath/carl9170/built-in.o
  CC      drivers/net/wireless/ath/main.o
  CC      drivers/net/wireless/ath/regd.o
  SHIPPED drivers/tty/vt/defkeymap.c
  CC      drivers/tty/vt/consolemap_deftbl.o
  CC      drivers/net/wireless/ath/hw.o
  CC      drivers/tty/vt/defkeymap.o
  LD      drivers/tty/vt/built-in.o
  LD      drivers/tty/built-in.o
  CC      drivers/uio/uio.o
  LD      drivers/net/wireless/bcmdhd/bcmdhd.o
  CC      drivers/net/wireless/ath/key.o
  LD      drivers/net/wireless/bcmdhd/built-in.o
  CC      drivers/uio/msm_sharedmem.o
  CC      drivers/usb/class/cdc-acm.o
  CC      drivers/usb/core/usb.o
  LD      drivers/uio/built-in.o
  CC      drivers/video/fb_notify.o
  CC      drivers/video/fbmem.o
  LD      drivers/net/wireless/ath/ath.o
  LD      drivers/net/wireless/ath/built-in.o
make[2]: *** [scripts/Makefile.build:443: drivers/net/wireless] Error 2
make[1]: *** [scripts/Makefile.build:443: drivers/net] Error 2
make[1]: *** Waiting for unfinished jobs....
  CC      drivers/video/fbmon.o
  CC      drivers/usb/core/hub.o
  CC      drivers/usb/class/usblp.o
  CC      drivers/video/fbcmap.o
  CC      drivers/usb/class/cdc-wdm.o
  CC      drivers/video/fbsysfs.o
  CC      drivers/video/modedb.o
  CC      drivers/video/fbcvt.o
  LD      drivers/usb/class/built-in.o
  CC      drivers/usb/dwc3/core.o
  CC      drivers/video/backlight/backlight.o
  CC      drivers/video/console/dummycon.o
  LD      drivers/video/console/built-in.o
  CC      drivers/usb/core/hcd.o
  CC      drivers/video/msm/mdss/dsi_v2.o
  CC      drivers/usb/dwc3/host.o
  CC      drivers/video/backlight/lm3630_bl.o
  CC      drivers/video/msm/mdss/dsi_host_v2.o
  CC      drivers/usb/dwc3/gadget.o
  LD      drivers/video/backlight/built-in.o
  CC      drivers/video/msm/mdss/dsi_io_v2.o
  LD      drivers/video/omap2/displays/built-in.o
  LD      drivers/video/omap2/built-in.o
  CC      drivers/video/msm/mdss/dsi_panel_v2.o
  CC      drivers/usb/core/urb.o
  CC      drivers/video/msm/mdss/mdss_dsi.o
  CC      drivers/usb/gadget/udc-core.o
  CC      drivers/usb/dwc3/ep0.o
  CC      drivers/usb/core/message.o
  CC      drivers/video/msm/mdss/mdss_dsi_host.o
  CC      drivers/usb/gadget/android.o
  CC      drivers/usb/dwc3/dwc3_otg.o
  CC      drivers/usb/dwc3/debugfs.o
  CC      drivers/video/msm/mdss/mdss_dsi_panel.o
  CC      drivers/usb/core/driver.o
  CC      drivers/usb/dwc3/dwc3-msm.o
  CC      drivers/video/msm/mdss/msm_mdss_io_8974.o
  CC      drivers/usb/core/config.o
  CC      drivers/video/msm/mdss/mdss_mdp.o
  CC      drivers/usb/core/file.o
  CC      drivers/usb/core/buffer.o
  LD      drivers/usb/dwc3/dwc3.o
  LD      drivers/usb/dwc3/built-in.o
  CC      drivers/usb/host/xhci.o
  CC      drivers/usb/core/sysfs.o
  CC      drivers/video/msm/mdss/mdss_mdp_ctl.o
  CC      drivers/usb/core/endpoint.o
  CC      drivers/usb/core/devio.o
  CC      drivers/video/msm/mdss/mdss_mdp_pipe.o
  CC      drivers/usb/host/xhci-mem.o
  CC      drivers/video/msm/mdss/mdss_mdp_util.o
  CC      drivers/usb/core/notify.o
  CC      drivers/video/msm/mdss/mdss_mdp_pp.o
  CC      drivers/usb/core/generic.o
  CC      drivers/usb/host/xhci-ring.o
  CC      drivers/usb/core/quirks.o
  CC      drivers/usb/core/devices.o
  LD      drivers/usb/core/usbcore.o
  LD      drivers/usb/core/built-in.o
  CC      drivers/video/msm/mdss/mdss_mdp_intf_video.o
  CC      drivers/usb/host/xhci-hub.o
  CC      drivers/video/msm/mdss/mdss_mdp_intf_cmd.o
  CC      drivers/video/msm/mdss/mdss_mdp_intf_writeback.o
  CC      drivers/usb/host/xhci-dbg.o
  CC      drivers/usb/gadget/f_hid.o
  CC      drivers/video/msm/mdss/mdss_mdp_rotator.o
  CC      drivers/usb/host/xhci-plat.o
  CC      drivers/video/msm/mdss/mdss_mdp_overlay.o
  CC      drivers/video/msm/mdss/mdss_mdp_wb.o
  LD      drivers/usb/host/xhci-hcd.o
  LD      drivers/usb/host/built-in.o
  CC      drivers/usb/misc/trancevibrator.o
  LD      drivers/usb/misc/built-in.o
  CC      drivers/usb/otg/otg.o
  LD      drivers/usb/otg/built-in.o
  CC      drivers/usb/serial/usb-serial.o
  CC      drivers/video/msm/mdss/mdp3.o
  CC      drivers/video/msm/mdss/mdp3_dma.o
  CC      drivers/video/msm/mdss/mdp3_ctrl.o
  CC      drivers/usb/serial/generic.o
  CC      drivers/video/msm/mdss/mdp3_ppp.o
  CC      drivers/video/msm/mdss/mdp3_ppp_hwio.o
  CC      drivers/usb/serial/bus.o
  CC      drivers/video/msm/mdss/mdp3_ppp_data.o
  CC      drivers/usb/serial/option.o
  CC      drivers/video/msm/mdss/mdss_debug.o
  CC      drivers/video/msm/mdss/mdss_edp.o
  CC      drivers/usb/serial/sierra.o
  CC      drivers/video/msm/mdss/mdss_edp_aux.o
  CC      drivers/video/msm/mdss/mdss_io_util.o
  CC      drivers/usb/serial/usb_wwan.o
  CC      drivers/video/msm/mdss/mdss_hdmi_tx.o
  CC      drivers/video/msm/mdss/mdss_hdmi_util.o
  LD      drivers/usb/serial/usbserial.o
  LD      drivers/usb/serial/built-in.o
  CC      drivers/video/msm/mdss/mdss_hdmi_edid.o
  CC      drivers/video/msm/mdss/mdss_hdmi_hdcp.o
  CC      drivers/video/msm/mdss/mdss_hdmi_cec.o
  CC      drivers/video/msm/mdss/mdss_wb.o
  CC      drivers/video/msm/mdss/mdss_fb.o
  CC      drivers/video/msm/mdss/lcd_notify.o
  CC      drivers/video/msm/mdss/mdss_mdp_kcal_ctrl.o
  LD      drivers/video/msm/mdss/mdss-mdp3.o
  LD      drivers/video/msm/mdss/mdss-mdp.o
  LD      drivers/video/msm/mdss/dsi-v2.o
  LD      drivers/video/msm/mdss/mdss-dsi.o
  CC      drivers/usb/storage/alauda.o
  CC      drivers/usb/storage/cypress_atacb.o
  CC      drivers/usb/storage/datafab.o
  CC      drivers/usb/storage/ene_ub6250.o
  LD      drivers/video/msm/mdss/built-in.o
  LD      drivers/video/msm/built-in.o
  CC      drivers/video/slimport/slimport.o
  CC      drivers/usb/storage/freecom.o
  CC      drivers/usb/storage/isd200.o
  CC      drivers/usb/storage/jumpshot.o
  CC      drivers/video/slimport/slimport_tx_drv.o
  CC      drivers/usb/storage/karma.o
  CC      drivers/usb/storage/onetouch.o
  CC      drivers/usb/storage/sddr09.o
  CC      drivers/usb/storage/sddr55.o
  LD      drivers/video/slimport/built-in.o
  CC      drivers/video/cfbfillrect.o
  CC      drivers/usb/storage/shuttle_usbat.o
  CC      drivers/video/cfbcopyarea.o
  LD      drivers/usb/gadget/g_android.o
  LD      drivers/usb/gadget/built-in.o
  CC      drivers/usb/usb-common.o
  CC      drivers/video/cfbimgblt.o
  CC      drivers/usb/storage/scsiglue.o
  LD      drivers/video/fb.o
  CC      drivers/usb/storage/protocol.o
  CC      drivers/usb/storage/transport.o
  LD      drivers/video/built-in.o
  CC      drivers/usb/storage/usb.o
  CC      drivers/usb/storage/initializers.o
  CC      drivers/usb/storage/sierra_ms.o
  CC      drivers/usb/storage/option_ms.o
  CC      drivers/usb/storage/usual-tables.o
  LD      drivers/usb/storage/usb-storage.o
  LD      drivers/usb/storage/ums-alauda.o
  LD      drivers/usb/storage/ums-cypress.o
  LD      drivers/usb/storage/ums-datafab.o
  LD      drivers/usb/storage/ums-eneub6250.o
  LD      drivers/usb/storage/ums-freecom.o
  LD      drivers/usb/storage/ums-isd200.o
  LD      drivers/usb/storage/ums-jumpshot.o
  LD      drivers/usb/storage/ums-karma.o
  LD      drivers/usb/storage/ums-onetouch.o
  LD      drivers/usb/storage/ums-sddr09.o
  LD      drivers/usb/storage/ums-sddr55.o
  LD      drivers/usb/storage/ums-usbat.o
  LD      drivers/usb/storage/built-in.o
  LD      drivers/usb/built-in.o
make: *** [Makefile:952: drivers] Error 2
`

**Comment by Darkar25** [comment]
I am not getting this error right now but i have similar environment and i may encounter this issue...so i am interested in solution too

**Comment by Darkar25** [comment]
Okay,i got it compiling...check my fork if you still interested...but i cant make it work on my device for some reason...
